lib/array.js

```diff
@@ -151,7 +151,8 @@ const filter = (items, fn, done) => {
   }
 };
 
-const EMPTY_ARR = 'Metasync: reduce of empty array with no initial value';
+const REDUCE_EMPTY_ARR =
```

Comment:
This and change and the change in the doc comment are semi-relevant, I am fine with having them in this PR, though I would like to extract them into a separate commit.

Reply:
I will leave 3 commits then:
1. reduceRight and tests for it.
2. Fix of the reduce.
3. Change of the comments and errors.

Reply:
Done